arwen/host/arwen.go

```diff
@@ -311,69 +306,55 @@ func (host *vmHost) RunSmartContractCreate(input *vmcommon.ContractCreateInput)
 	host.mutExecution.RLock()
 	defer host.mutExecution.RUnlock()
 
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*2)
+	defer cancel()
+
 	log.Trace("RunSmartContractCreate begin", "len(code)", len(input.ContractCode), "metadata", input.ContractCodeMetadata)
 
-	try := func() {
+	done := make(chan interface{}, 1)
```

Comment:
```GO
done := make(chan struct{})
```
and on L317
```GO
close(done)
```

It is always safer to close the channel instead of writing a struct to it. Because someone can easily alter the code to unbuffer the chan, all tests pass but we end up with go routines leakage.

valid for all occurrences

Reply:
done